portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -28,14 +28,28 @@
 public class DataServlet extends HttpServlet {
 
   private ArrayList<String> shows;
-
-  @Override
+ 
+  /*@Override
   public void init(){
       shows = new ArrayList<String>();
       shows.add("The Good Doctor");
       shows.add("Grey\'s Anatomy");
       shows.add("House");
       shows.add("Code Black");
+  }*/
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      shows = new ArrayList<String>();
+      String text = request.getParameter("text-input");
+      
+      shows.add(text);
+
+      Gson gson = new Gson();
+      String json = gson.toJson(shows);
+
+      response.setContentType("text/html;");
+      response.getWriter().println(json);
   }
```

Comment:
So you actually don't have to comment out the doGet() function in order to add the doPost() one! You need to keep the doGet() function since it's the one that will return the value of your `shows` variable to the frontend.

All you need to do is to remove the hardcoded values for `shows`, which you already did. Please restore your doGet() function by removing the comment slashes :)